xdsl/transforms/convert_scf_to_openmp.py

```diff
@@ -43,7 +44,8 @@ def match_and_rewrite(self, loop: scf.ParallelOp, rewriter: PatternRewriter, /):
             regions=[Region(Block())],
             operands=[[], [], [], [], [], []],
         )
-        with ImplicitBuilder(parallel.region):
+        rewriter.insertion_point = InsertPoint.at_end(parallel.region.block)
+        with ImplicitBuilder(rewriter):
```

Comment:
I don't think I'm a fan of this API. Does the user need to reset the insertion point of the rewriter after doing this?

Reply:
I agree the API is not optimal.
Currently, what we need to use `ImplicitBuilder` is:
* set the insertion point on the rewriter
* Create the ImplicitBuilder using the rewriter

We don't need to reset it after though.

What would you think of `ImplicitBuilder(rewriter, at=InsertPoint.at_end(...))`, which would set the insertion point just for that region of code?